src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
+
+impl From<NulError> for NameToIndexError {
+    fn from(_: NulError) -> NameToIndexError {
+        NameToIndexError::NulError
+    }
+}
+
+impl From<io::Error> for NameToIndexError {
+    fn from(e: io::Error) -> NameToIndexError {
+        NameToIndexError::IOError(e)
+    }
+}
+
+extern {
+    fn if_nametoindex(ifname: *const c_char) -> c_uint;
```

Comment:
use this from libc instead of making our own exter: http://rust-lang-nursery.github.io/libc/x86_64-unknown-linux-gnu/libc/fn.if_nametoindex.html